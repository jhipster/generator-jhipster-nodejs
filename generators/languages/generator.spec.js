import { beforeAll, describe, expect, it } from 'vitest';

import { defaultHelpers as helpers, result } from 'generator-jhipster/testing';

const SUB_GENERATOR = 'languages';
const BLUEPRINT_NAMESPACE = `jhipster:${SUB_GENERATOR}`;

describe('SubGenerator languages of nodejs JHipster blueprint', () => {
  describe('run', () => {
    beforeAll(async function () {
      await helpers
        .run(BLUEPRINT_NAMESPACE)
        .withJHipsterConfig()
        .withOptions({
          ignoreNeedlesError: true,
          blueprint: ['nodejs'],
        })
        .withJHipsterLookup()
        .withParentBlueprintLookup();
    });

    it('should succeed', () => {
      expect(result.getStateSnapshot()).toMatchSnapshot();
    });
  });
});
