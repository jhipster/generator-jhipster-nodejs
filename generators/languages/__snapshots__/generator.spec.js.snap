// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SubGenerator languages of nodejs JHipster blueprint > run > should succeed 1`] = `
{
  ".yo-rc.json": {
    "stateCleared": "modified",
  },
  "client/src/i18n/en/activate.json": {
    "stateCleared": "modified",
  },
  "client/src/i18n/en/error.json": {
    "stateCleared": "modified",
  },
  "client/src/i18n/en/global.json": {
    "stateCleared": "modified",
  },
  "client/src/i18n/en/home.json": {
    "stateCleared": "modified",
  },
  "client/src/i18n/en/login.json": {
    "stateCleared": "modified",
  },
  "client/src/i18n/en/password.json": {
    "stateCleared": "modified",
  },
  "client/src/i18n/en/register.json": {
    "stateCleared": "modified",
  },
  "client/src/i18n/en/reset.json": {
    "stateCleared": "modified",
  },
  "client/src/i18n/en/sessions.json": {
    "stateCleared": "modified",
  },
  "client/src/i18n/en/settings.json": {
    "stateCleared": "modified",
  },
  "client/src/i18n/en/user-management.json": {
    "stateCleared": "modified",
  },
  "client/src/i18n/fr/activate.json": {
    "stateCleared": "modified",
  },
  "client/src/i18n/fr/error.json": {
    "stateCleared": "modified",
  },
  "client/src/i18n/fr/global.json": {
    "stateCleared": "modified",
  },
  "client/src/i18n/fr/home.json": {
    "stateCleared": "modified",
  },
  "client/src/i18n/fr/login.json": {
    "stateCleared": "modified",
  },
  "client/src/i18n/fr/password.json": {
    "stateCleared": "modified",
  },
  "client/src/i18n/fr/register.json": {
    "stateCleared": "modified",
  },
  "client/src/i18n/fr/reset.json": {
    "stateCleared": "modified",
  },
  "client/src/i18n/fr/sessions.json": {
    "stateCleared": "modified",
  },
  "client/src/i18n/fr/settings.json": {
    "stateCleared": "modified",
  },
  "client/src/i18n/fr/user-management.json": {
    "stateCleared": "modified",
  },
}
`;
